requirements.txt

```diff
@@ -1,8 +1,8 @@
-dbt-core~=0.18.0
+git+https://github.com/fishtown-analytics/dbt.git@dev/kiyoshi-kuromiya
 pyodbc>=4.0.27
 azure-identity>=1.4.0
 black~=20.8b1
-pytest-dbt-adapter~=0.3.0
+pytest-dbt-adapter~=0.4.0
```

Comment:
I think this change is necessary for `v0.19.0`? Anyways, I changed it, and now we're getting a new error @mikaelene. Perhaps we need snapshot_check_strategies` after all?

> An action of type 'WHEN MATCHED' cannot appear more than once in a 'UPDATE' clause of a MERGE statement.

[This SO answer](https://stackoverflow.com/questions/34977599/why-when-matched-cannot-appear-more-than-once-in-a-update-clause-of-a-merge-s) says if we have two `when matched` one must be a delete...

### `snapshot.dbt_test_project.ts_snapshot` (the problem query)
```sql
merge into "**********************"."dbt_test_azure_sql_210114081830263174238692"."ts_snapshot" as DBT_INTERNAL_DEST
using "**********************"."dbt_test_azure_sql_210114081830263174238692"."#ts_snapshot__dbt_tmp" as DBT_INTERNAL_SOURCE
on DBT_INTERNAL_SOURCE.dbt_scd_id = DBT_INTERNAL_DEST.dbt_scd_id

when matched
    and DBT_INTERNAL_DEST.dbt_valid_to is null
    and DBT_INTERNAL_SOURCE.dbt_change_type = 'update'
    then update
    set dbt_valid_to = DBT_INTERNAL_SOURCE.dbt_valid_to

when matched
    and DBT_INTERNAL_DEST.dbt_valid_to is null
    and DBT_INTERNAL_SOURCE.dbt_change_type = 'delete'
    then update
    set dbt_valid_to = DBT_INTERNAL_SOURCE.dbt_valid_to

when not matched
    and DBT_INTERNAL_SOURCE.dbt_change_type = 'insert'
    then insert ("id", "name", "some_date", "dbt_updated_at", "dbt_valid_from", "dbt_valid_to", "dbt_scd_id")
    values ("id", "name", "some_date", "dbt_updated_at", "dbt_valid_from", "dbt_valid_to", "dbt_scd_id")
;
```

Reply:
@swanderz Right move to upgrade to `pytest-dbt-adapter v0.4.0`.

As for the snapshot `merge` code, Mikael actually already fixed it here: https://github.com/fishtown-analytics/dbt/pull/3004. That code will be included in dbt v0.19.0-rc2.

Even if the default code _didn't_ work, it's a much more straightforward override of a more modular dispatched macro: `synapse__snapshot_merge_sql`.